config/default/kustomization.yaml

```
@@ -52,6 +51,40 @@ patches:
 # Expose port used by the metrics service
 - path: manager_metrics_patch.yaml
 
+# Add component labels to services
+- patch: |-
+    apiVersion: v1
+    kind: Service
+    metadata:
+      name: webhook-service
+      namespace: system
+      labels:
+        app.kubernetes.io/component: webhook-service
+- patch: |-
+    apiVersion: v1
+    kind: Service
+    metadata:
+      name: visibility-server
+      namespace: kueue-system
+      labels:
+        app.kubernetes.io/component: visibility-service
+- patch: |-
+    apiVersion: v1
+    kind: Service
+    metadata:
+      name: controller-manager-metrics-service
+      namespace: system
+      labels:
+        app.kubernetes.io/component: controller-manager-metrics-service
```

```suggestion
        app.kubernetes.io/component: metrics-service
```
I want to propose simpler name here. We don't need to rename the service name in this PR, just the label. The name of the labels does not need to match the service name exactly anyway as the service names have the "kueue-" prefix anyway. wdyt? cc @tenzen-y

---

That makes sense.
In that case, the service for keueu external-controller-manager could have `app.kubernetes.io/name: kueue-external-manager` and `app.kubernetes.io/instance: metrics-service` label.